api/app/src/main/kotlin/packit/service/OrderlyRunnerClient.kt

```diff
@@ -0,0 +1,63 @@
+package packit.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.core.ParameterizedTypeReference
+import org.springframework.http.*
+import org.springframework.stereotype.Service
+import org.springframework.web.client.RestTemplate
+import packit.AppConfig
+import packit.exceptions.PackitException
+import packit.model.ServerResponse
+import packit.model.dto.OrderlyRunnerVersion
+
+interface OrderlyRunner
+{
+    fun getVersion(): OrderlyRunnerVersion
+}
+
+@Service
+class OrderlyRunnerClient(appConfig: AppConfig) : OrderlyRunner
+{
+    val baseUrl: String = appConfig.orderlyRunnerUrl
+    private val restTemplate = RestTemplate()
+
+    override fun getVersion(): OrderlyRunnerVersion
+    {
+        return getEndpoint("/")
+    }
+
+    private inline fun <reified T> getEndpoint(urlFragment: String): T
+    {
+        val url = "$baseUrl/$urlFragment"
+        log.debug("Fetching {}", url)
```

Comment:
Rogue debug?

Reply:
just gonna tag anmol in the kotlin related comments so he doesnt miss them, @absternator